Admin api

Nyutonyka opened the pull request:
An error appears on startup: Parameter 0 of constructor in com.store.service.impl.ProductAdminServiceImpl required a bean of type 'com.store.repository.ProductAdminRepository' that could not be found.

MariaTarasiuk commented on src/main/java/com/store/controller/ProductAdminController.java:
```diff
@@ -0,0 +1,32 @@
+
+    @PostMapping
+    public ResponseEntity<Product> addProduct(@RequestBody ProductAdminDto productAdminDto) {
+        Product product = productAdminService.addProduct(productAdminDto);
```
It is a good approach to isolate DB entity from any external client.
FE knows only ProductAdminDto. Product - it is our internal entity class. If FE sends ProductAdminDto, please return then ResponseEntity<ProductAdminDto>. And always return only ProductAdminDto, not entity class. You need to create mapping from Product to ProductAdminDto.

Nyutonyka replied:
Here I did not understand. Everything is correct or not. If not, can you tell me how to fix it?

MariaTarasiuk replied:
Your method in Controller should return ResponseEntity<ProductAdminDto>. What means addProduct() in productAdminService should return ProductAdminDto. Query to DB returns Product, so you need to convert Product to ProductAdminDto in service.

Nyutonyka replied:
I reworked it a bit and I don't return ResponseEntity. We will return the id of the created product. Everything works, the product is added to the database

MariaTarasiuk commented on src/main/java/com/store/service/impl/ProductAdminServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+        product.setCategory(productAdminDto.getCategory());
+        product.setDescription(productAdminDto.getDescription());
+        product.setQuantity(productAdminDto.getQuantity());
+        return productAdminRepository.save(product);
```
To make this code more clear you can use @Builder from lombok. This code also will work, but if you wish, you can change it.

MariaTarasiuk commented on src/main/java/com/store/controller/ProductAdminController.java:
```diff
@@ -0,0 +1,32 @@
+public class ProductAdminController {
+
+    private final ProductAdminServiceImpl productAdminService;
+
```
If you created an interface it makes sense to inject interface here

Nyutonyka replied:
Corrected

MariaTarasiuk commented on src/main/java/com/store/service/ProductAdminService.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface ProductAdminService {
+
+    Product addProduct(ProductAdminDto productAdminDto);
```
This interface is not needed here, but if you want you can leave it

Nyutonyka replied:
I will leave the interfaces

MariaTarasiuk commented:
> 

I don't understand why you added exclude here: @SpringBootApplication(exclude = {DataSourceAutoConfiguration.class })
Try to delete exclude, maybe it will fix a problem.

MariaTarasiuk commented:
As I see, this PR is in main. But as I remember we decided main - only for tested tasks. Dev - for development and testing

Nyutonyka commented:
@SpringBootApplication(exclude = {DataSourceAutoConfiguration.class }) I added it because there was an error without BD connection. Now I deleted this annotation and all the work

Regarding the branch, can we try to leave it as we initially talked?

Nyutonyka commented:
Added validation, checked all work

MariaTarasiuk commented on src/main/java/com/store/dto/ProductAdminDto.java:
```diff
@@ -0,0 +1,34 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductAdminDto {
+    @NotNull(message = "Article is required")
```
you need to add one more field here with Product id. It is a primary key in DB.
